Services/Certificate/classes/Placeholder/Values/class.ilExercisePlaceHolderValues.php

```diff
@@ -1,70 +1,29 @@
-<?php
+<?php declare(strict_types=1);
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilExercisePlaceHolderValues implements ilCertificatePlaceholderValues
 {
-    /**
-     * @var ilLanguage|null
-     */
-    private $language;
-
-    /**
-     * @var ilDefaultPlaceholderValues|null
-     */
-    private $defaultPlaceHolderValuesObject;
-
-    /**
-     * @var ilCertificateLPMarksHelper|null
-     */
-    private $lpMarksHelper;
-
-    /**
-     * @var ilCertificateObjectHelper|null
-     */
-    private $objectHelper;
-
-    /**
-     * @var ilCertificateExerciseMembersHelper|null
-     */
-    private $exerciseMembersHelper;
+    private ?ilLanguage $language;
+    private ?ilDefaultPlaceholderValues $defaultPlaceHolderValuesObject;
+    private ?ilCertificateLPMarksHelper $lpMarksHelper;
+    private ?ilCertificateObjectHelper $objectHelper;
+    private ?ilCertificateExerciseMembersHelper $exerciseMembersHelper;
+    private ?ilCertificateLPStatusHelper $lpStatusHelper;
+    private ?ilCertificateUtilHelper $utilHelper;
+    private ?ilCertificateDateHelper $dateHelper;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.